src/Initializer/time_stepping/LtsLayout.cpp

```diff
@@ -73,27 +78,26 @@ seissol::initializers::time_stepping::LtsLayout::~LtsLayout() {
   delete[] m_plainCopyRegions;
 }
 
-void seissol::initializers::time_stepping::LtsLayout::setMesh( const MeshReader &i_mesh ) {
+void seissol::initializers::time_stepping::LtsLayout::setMesh( const seissol::geometry::MeshReader &i_mesh ) {
   // TODO: remove the copy by a pointer once the mesh stays constant
   m_cells = i_mesh.getElements();
   m_fault = i_mesh.getFault();
 
-  m_cellTimeStepWidths = new double[       m_cells.size() ];
   m_cellClusterIds     = new unsigned int[ m_cells.size() ];
 
   // initialize with invalid values
   for (unsigned int l_cell = 0; l_cell < m_cells.size(); ++l_cell) {
-    m_cellTimeStepWidths[l_cell] = std::numeric_limits<double>::min();
     m_cellClusterIds[l_cell] = std::numeric_limits<unsigned int>::max();
   }
-}
 
-void seissol::initializers::time_stepping::LtsLayout::setTimeStepWidth( unsigned int i_cellId,
-                                                                        double       i_timeStepWidth ) {
-  if( i_cellId >= m_cells.size() ) logError() << "cell id >= mesh size: " << i_cellId << m_cells.size() << "aborting";
+  auto& seissolParams = seissol::SeisSol::main.getSeisSolParameters();
 
-  // set time step width
-  m_cellTimeStepWidths[i_cellId] = i_timeStepWidth;
+  // compute timesteps
+  auto timesteps = seissol::initializer::computeTimesteps(seissolParams.timestepping.cfl, seissolParams.timestepping.maxTimestepWidth, seissolParams.model.materialFileName,
+        seissol::initializers::CellToVertexArray::fromMeshReader(i_mesh));
+  
+  m_cellTimeStepWidths = new double[m_cells.size()];
```

Comment:
is there a good reason not to use a std::vector?

Reply:
There is actually none to not use it... So it's done now. :) Refactoring more of `LTSLayout` will stay a task for the future though, I'd guess.